### ISSUE TITLE
Dell USB adapter does not work properly

### ISSUE THREAD

**Issue by domak** [advancingu/XPS13Linux#6]
This adapter connects an USB 3 to USB 2/Ethernet/VGA/HDMI ports (http://accessories.us.dell.com/sna/productdetail.aspx?c=us&l=en&s=dhs&cs=19&sku=470-abhh).
- USB 2 and Ethernet works
- HDMI does not work and it seems that no driver will be developed (http://bartongeorge.net/2015/04/09/4th-gen-dell-xps-13-developer-edition-available/#comment-93298)
- VGA have not been tested

**Comment by dchambers** [comment]
@domak, this is a long standing Linux thing (not Dell specific), and will never be fixed. However, with the latest skylake and broadwell laptops plus the [DisplayPort MST work by David Arlie](http://airlied.livejournal.com/79657.html) these USB 3 hubs should soon become redundant anyway, as daisy chained DisplayPort becomes viable directly from the laptop.

Mario Limonciello [has even hinted](http://bartongeorge.net/2015/04/09/4th-gen-dell-xps-13-developer-edition-available/#comment-92596) that this may work on Ubuntu 15.04 even. Perhaps you can try it, and let us know if it works or not?

**Comment by ollyhayes** [comment]
I'm the person Mario was replying to. My two monitors don't support daisy chaining so I've bought a star-tech MRT hub. Still waiting for another adaptor to test both monitors at once but it was definitely driving one monitor through the hub, though I guess that's not surprising.

Once my second adaptor arrives I'll write back here whether it works or not.

I'm using Ubuntu 15.04.

**Comment by dchambers** [comment]
Hi @ollyhayes, thanks for the info, and I look forward to hearing how you get on with that hub. BTW, I'm guessing this is the [star-tech MRT hub](http://uk.startech.com/AV/Displayport-Converters/Triple-Head-DisplayPort-Multi-Monitor-MST-Hub~MSTDP123DP) you are referring to? Very handy as I didn't know anything like this existed.

**Comment by ollyhayes** [comment]
Yeah but with a mini display port input, sorry I would have linked but was in a rush earlier.

This is the exact one I bought: http://www.amazon.co.uk/gp/product/B00JLRBC7S

Bit expensive but cheaper than buying all new monitors!

**Comment by ollyhayes** [comment]
Adapter has arrived, works perfectly with two external 1280/1024 monitors (3 total). Though it did lock up first time I tried, I had to plug the monitors into the MRT hub one at a time and then it worked.

I'm going to get another adaptor so I can plug it in to my TV too and see if it can drive three monitors, if so I'll probably buy a third 1280x1024 monitor. I'll post back here when I have more info!

**Comment by dchambers** [comment]
Nice work @ollyhayes, that's really good to know. I think the thing about having to plug the monitors in one at a time seems quite normal, as I heard Windows users saying the same kind of thing within some Amazon feedback I was looking at.

Please let us know how you get on with 3 external monitors too...

**Comment by ollyhayes** [comment]
Couldn't get it to work with three I'm afraid, wouldn't even recognise the third monitor, even if I turned the built in display off.

It all seemed really buggy, the laptop locked up three or four times during my attempts, the pointer would move around but nothing would respond to clicking, and caps lock wouldn't turn the little light on or off. I had to hard reset by holding down the power button. Even with only two monitors attached it was still locking up sporadically like I mentioned before.

I think I'll maybe try with the 4.0 kernel and see if that makes a difference.

**Comment by dchambers** [comment]
@ollyhayes, did you ever try the 4.0 kernel, and if so did it help make the two monitor set-up more reliable?

**Comment by ollyhayes** [comment]
It's still on my to-do list! Been going away on the weekends a lot recently so it might be another week or two before I get a chance, I'll definitely report back when I finally get a chance though.

**Comment by dchambers** [comment]
:+1:

**Comment by advancingu** [comment]
For anyone who needs an adapter to HDMI, DVI, or VGA, we've bought a rebranded version of [this adapter](http://www.amazon.com/Cable-Matters%C2%AE-DisplayPort-Thunderbolt-Compatible/dp/B00ESM3ISM/ref=sr_1_fkmr1_2?ie=UTF8&qid=1438355727&sr=8-2-fkmr1&keywords=CSL+-+3in1+Mini+Display+Port) and it worked out of the box for 1920x1200 resolution via HDMI on Ubuntu 15.04 (haven't tested the other two connectors).

**Comment by jared-dominguez** [comment]
@advancingu That's an entirely different adapter. It's mini-DisplayPort to HDMI/DVI/VGA, not USB 3.0 to HDMI/DVI/VGA.

**Comment by ollyhayes** [comment]
I've finally gotten around to testing my [MST hub](http://www.amazon.co.uk/gp/product/B00JLRBC7S) with kernel 4.1.

Firstly I tried Linux Mint 17.2 and updated the kernel myself to 4.1. I could plug a single monitor in using the display port without the MST hub and it worked, but once I went through the hub I couldn't get event a single monitor out of it. Having it plugged in on startup prevented X from starting and plugging it in whilst running didn't do anything.

I tried getting kernel 4.1 to work on Ubuntu 15.04 but couldn't get it to boot so I gave up.

A few days ago I installed Antergos 2015.07.01 with Cinnamon. Support here for the MST is **much** more stable than on Ubuntu 15.04, I can hot plug as much as I like and never saw a lock-up or kernel panic. (Also seems to fix the kernel panic on suspend issue I was having with Ubuntu). Still couldn't get 3 external monitors to work though, it would only recognise 2 max, even if the built in display was disabled.

I may get Windows 10 in the next few months so I'll try again with that.

**Comment by antonmry** [comment]
Someone has tested the DisplayLink propietary driver?

http://downloads.displaylink.com/releasenotes/DisplayLink_Ubuntu_1.0.68_release-note.txt
http://www.displaylink.com/downloads/ubuntu.php

It looks very interesting. I am using Fedora, no Ubuntu, so I cann't use it yet.

**Comment by domak** [comment]
It works but you have to reinstall the driver each time, reconnect the hdmi cable after that and go to the display menu of system settings to detect display.
I use it on a tv with 1360x768 resolution and the menus are just too bigs (except ubuntu launcher who is displayed correctly). I didn't tried to tweak this because I'm using it only to watch movies).

**Comment by RCura** [comment]
@domak & @antonmry Seems the new version (1.0.138) is working way better : no need to reinstall the driver,  and it even works well when deconnection/reconnecting the cable, without having to reboot :)

**Comment by dchambers** [comment]
Well, I never thought I'd see the day! Better three years late than never I suppose :-)

**Comment by adomingues** [comment]
@domak  I can confirm what @RCura  said: new version (1.0.138)  is working. However, there is an issue with mouse flickering in one of the external monitors. My set-up is Ubuntu 14.04/Unity with kernel 3.19.0, and I have one monitor connected directly to the XPS13 via mini-DisplayPort, and another via DVI in the Dock. This bug has already been [reported](https://bugs.launchpad.net/ubuntu/+source/xorg/+bug/1278223), and there seem to be some workarounds for KDE.
